src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -574,9 +568,16 @@ public static TypeSymbol VisitType<T>(this TypeSymbol type, Func<TypeSymbol, T,
                     case TypeKind.Struct:
                     case TypeKind.Interface:
                     case TypeKind.Delegate:
-                        foreach (var typeArg in ((NamedTypeSymbol)current).TypeArgumentsNoUseSiteDiagnostics)
+                        // Consider: should tuple be its own TypeKind?
```

Comment:
We would need to special case a lot more scenarios if tuple had its own TypeKind. 
Right now it behaves like a Struct everywhere where its "tuple" nature is uninteresting. There are many cases like that since for CLR tuples are just structs.